src/libcore/convert.rs

```diff
@@ -30,6 +50,10 @@ use marker::Sized;
 ///
 /// [book]: ../../book/borrow-and-asref.html
 ///
+/// **Note:** these traits are for trivial conversion. **They must not fail**. If
```

Comment:
Ah no panics are still "unacceptable" but not in the sense that something like unsafety will arise. It's just unidiomatic and can lead to bugs, so it's rather **strongly recommended** that conversions don't panic.

The wording currently implies to me that something very bad will happen if a panic happens during a conversions, but it should be pretty normal in terms of the outcome (the code just panics). It's just the standard worry about a function panicking.